opennlp-tools/src/main/java/opennlp/tools/ml/AbstractEventTrainer.java

```diff
@@ -92,8 +107,8 @@ public final MaxentModel train(ObjectStream<Event> events) throws IOException {
 
     MaxentModel model = doTrain(indexer);
```

Comment:
Before we call the doTrain(), we still need to check to make sure the training parameters are valid.  The isValid() method in AbstractEventTrainer only check if the dataindexer is 1 or two pass, which is currently useful.  However, every trainer should override the isValid() method with a more specific method.  Either way, if you train with indexedData, you should still check if the parameters are valid.